### ISSUE TITLE
Firefox plugin won't connect to KeePassXC

### ISSUE THREAD

**Issue by ntomac** [keepassxreboot/keepassxc-browser#1342]
## Expected Behavior
I followed the process to connect firefox plugin to the desktop KeePassXC.
Browser integration is active.
When reloading the plugin it should connect.

## Current Behavior
Reloading the plugin gives the error message : "key swap failed" (in french "l'échange de clé a échoué").

## Possible Solution
I might forgot something ?

I already managed to set it on an another computer few months ago but I think i did the same

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Debug info

KeePassXC - unknown
KeePassXC-Browser - 1.7.8.1
Operating system: Linux x86_64
Browser: Mozilla Firefox 89.0

**Comment by varjolintu** [comment]
How is your Firefox installed? From repository? Or is it a snap/flatpak package?

**Comment by ntomac** [comment]
> How is your Firefox installed? From repository? Or is it a snap/flatpak package?

Hi varjolintu, it was installed natively on my kubuntu. I would say it's from repository then but i don't know how to check.

**Comment by varjolintu** [comment]
`about:support` should show you information about the application binary.

**Comment by ntomac** [comment]
i went to `about:support` in firefox, i have the menu, but i don't know about application binary.

At this moment all i can tell is that the line "application binary" shows : `/usr/lib/firefox/firefox`

**Comment by varjolintu** [comment]
How is your KeePassXC installed? Can you try the official AppImage?

**Comment by varjolintu** [comment]
> I'm experiencing the same issue using LibreWolf (It's a FF fork). It used to work just fine, but at some point after updating KPXC the extension would no longer connect (This was probably a few months ago).

Check the Native Messaging script. The path might have been changed.

**Comment by SteenSchutt** [comment]
Thanks. I had a suspicion that it was that, but couldn't really figure it out after playing around with it a bit. Copying `~/.mozilla/native-messaging-hosts` to `~/.librewolf/native-messaging-hosts` and changing the config location under Browser Integration -> Advanced to the `~/.librewolf/native-messaging-hosts` directory.

**Comment by ntomac** [comment]
Hi everybody

I finally got help from a friend. We compared with the keepass version on my other computer and we saw that the one where it worked was installed from my distribution repository, whereas the one with my problem was as a snap.
I removed the snap and installed from the repository and everything worked.

As SteenSchutt observed, we saw that `~/.mozilla/native-messaging-hosts` was not created when as a snap, but it was when from repository.

Thanks everybody for your help.
